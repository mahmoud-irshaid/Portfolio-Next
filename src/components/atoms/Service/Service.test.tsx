import { Service } from './Service';

import { render } from '@testing-library/react';

jest.mock('@lib/sanity', () => ({
	sanityClient: {
		fetch: jest.fn(),
	},
}));

describe('Service', () => {
	it('should render', () => {
		const defaultProps = {
			title: 'Service',
			subtitle: 'Some Subtitle',
			slug: 'service',
			body: [],
			image: 'https://via.placeholder.com/900x600',
		};

		const { container } = render(<Service {...defaultProps} />);

		expect(container.firstChild).toMatchSnapshot();
	});
});
