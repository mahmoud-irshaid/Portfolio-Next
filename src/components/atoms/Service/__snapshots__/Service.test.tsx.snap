// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service should render 1`] = `
<article
  id="service"
>
  <h3
    class="headline mt-20 text-xl md:text-2xl lg:text-4xl"
  >
    Service
  </h3>
  <div
    class="flex flex-col lg:gap-12 lg:flex-row"
  >
    <img
      alt="Illustration for Service"
      class="mt-8 px-12 lg:mt-0 lg:w-1/4 lg:px-0"
      data-nimg="1"
      decoding="async"
      height="600"
      loading="lazy"
      src="/_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F900x600&w=1920&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F900x600&w=1080&q=75 1x, /_next/image?url=https%3A%2F%2Fvia.placeholder.com%2F900x600&w=1920&q=75 2x"
      style="color: transparent;"
      width="900"
    />
    <div>
      <p
        class="mt-4 text-xl font-semibold md:text-xl lg:text-2xl"
      >
        Some Subtitle
      </p>
    </div>
  </div>
</article>
`;
